### PR TITLE
AD Snow example and classes

### LINKED ISSUE #15508: Convert Kobayashi solidification model to AD
## Reason
Demonstrate the use of AD in phase field. Permit performance comparisons.

## Design
Convert to AD, remove hand coded Jacobian.

## Impact
Demonstration case. Performance test.

### PR DESCRIPTION
**Pull request by dschwen**
Closes #15508

### CODE REVIEW

File: modules/phase_field/doc/content/bib/phase_field.bib [hunk 1]
```diff
@@ -144,3 +144,16 @@ @article{ClaytonKnap2015
...
 }
 
+@article{Kobayashi1993,
+title = "Modeling and numerical simulations of dendritic crystal growth",
+journal = "Physica D: Nonlinear Phenomena",
+volume = "63",
+number = "3",
+pages = "410 - 423",
+year = "1993",
+issn = "0167-2789",
```
Comment (aeslaughter) [inline modules/phase_field/doc/content/bib/phase_field.bib:154]:
```suggestion
pages = "410--423",
year = "1993",
issn = "0167--2789",
```

File: modules/phase_field/include/kernels/ADCoefCoupledTimeDerivative.h [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+{
+public:
+  static InputParameters validParams();
+
+  ADCoefCoupledTimeDerivative(const InputParameters & parameters);
+
+protected:
+  ADReal precomputeQpResidual() override;
+
+  const Real _coef;
```
Comment (aeslaughter) [inline modules/phase_field/include/kernels/ADCoefCoupledTimeDerivative.h:28]:
```suggestion
  const Real & _coef;
```

File: modules/phase_field/include/materials/ADInterfaceOrientationMaterial.h [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "Material.h"
+
+// Forward Declarations
```
Comment (aeslaughter) [inline modules/phase_field/include/materials/ADInterfaceOrientationMaterial.h:14]:
```suggestion
```

File: modules/phase_field/include/materials/ADInterfaceOrientationMaterial.h [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  ADInterfaceOrientationMaterial(const InputParameters & parameters);
+
+protected:
+  virtual void computeQpProperties();
+
+private:
+  Real _delta;
+  unsigned int _j;
+  Real _theta0;
+  Real _eps_bar;
```
Comment (aeslaughter) [inline modules/phase_field/include/materials/ADInterfaceOrientationMaterial.h:35]:
```suggestion
  const Real & _delta;
  const unsigned int & _j;
  const Real & _theta0;
  const Real & _eps_bar;
```

File: modules/phase_field/src/kernels/ADCoefCoupledTimeDerivative.C [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+#include "ADCoefCoupledTimeDerivative.h"
+
+registerMooseObject("PhaseFieldApp", ADCoefCoupledTimeDerivative);
+
+InputParameters
+ADCoefCoupledTimeDerivative::validParams()
+{
+  InputParameters params = ADCoupledTimeDerivative::validParams();
+  params.addClassDescription("Scaled time derivative Kernel that acts on a coupled variable");
+  params.addRequiredParam<Real>("coef", "Coefficient");
```
Comment (aeslaughter) [inline modules/phase_field/src/kernels/ADCoefCoupledTimeDerivative.C:19]:
```suggestion
  params.addRequiredParam<Real>("coef", "Scaling term applied to the time derivative.");
```

**Review by aeslaughter — CHANGES_REQUESTED** [review]

File: modules/phase_field/doc/content/bib/phase_field.bib (hunk 1 above)
Comment (aeslaughter) [reply modules/phase_field/doc/content/bib/phase_field.bib:154]: oops, that issn "fix" is wrong, don't do that little guy

**Comment by aeslaughter** [comment]
Are you still working on this?

**Comment by dschwen** [comment]
This was finished and merged in #15891

**Pull request closed without merging**